### PR TITLE
Implementing NEON-ized Adler32 checksum

### PR DESCRIPTION
**Pull request by Adenilson**
The checksum is calculated in the uncompressed PNG data and can be made much faster by using SIMD. Tests in ARMv8 yielded an improvement of about 3x (e.g. walltime was 350ms x 125ms for a 4096x4096 bytes executed 30 times).

This yields an improvement in image decoding in Chromium around 18% (see https://bugs.chromium.org/p/chromium/issues/detail?id=688601).

### CODE REVIEW

**Comment by Adenilson** [comment]
I tried to enable build by adding this: https://gist.github.com/Adenilson/1588e7448fa15a161584887a5b3ba04e

**Comment by Adenilson** [comment]
The output: https://gist.github.com/Adenilson/75880a60c70bc5cd8ecc12fa4e503a50

I wonder where I got to change the buildsystem to tell the compiler where to find the neon adler code?

File: adler32.c [hunk 1]
```diff
@@ -7,6 +7,10 @@
 
 #include "zutil.h"
 
+#ifdef __ARM_NEON__
```
Comment (Adenilson) [inline adler32.c]: Damn, forgot this one.

File: adler32.c [hunk 2]
```diff
@@ -7,6 +7,10 @@
 
 #include "zutil.h"
 
+#if (defined(__ARM_NEON__) || defined(__ARM_NEON))
+#include "neon_adler32.h"
```
Comment (mtl1979) [inline adler32.c]: This needs to be "adler32_neon.h"

File: adler32.c [hunk 3]
```diff
@@ -61,6 +65,10 @@ static uint32_t adler32_combine_(uint32_t adler1, uint32_t adler2, z_off64_t len
 
 /* ========================================================================= */
 uint32_t ZEXPORT adler32_z(uint32_t adler, const unsigned char *buf, size_t len) {
+#if (defined(__ARM_NEON__) || defined(__ARM_NEON))
+    return NEON_adler32(adler, buf, len);
```
Comment (mtl1979) [inline adler32.c]: Rename to "adler32_neon" to match other variants.

**Comment by Adenilson** [comment]
I'm assuming that when we are done with the review, we can simply rebase/squash all the commits in a single CL?

**Comment by mtl1979** [comment]
@Adenilson Squashing makes it cleaner when the pull request is about adding just a single feature.

**Comment by Adenilson** [comment]
It seems I got use the hacknslash7 branch, I'm closing this merge request and creating a new.

**Comment by mtl1979** [comment]
@Adenilson Just rebase the local branch, no need to make another merge request

**Pull request closed without merging**